### PR TITLE
[CIS-556] Make types following naming convention

### PR DESCRIPTION
**Pull request by evsaev**
**This PR**:
- updates type names in `StreamChatUI` to have `<LLC_model_name>` prefix the UI component serves
- creates aliases for generic UI components with `NoExtraData`
- adds some UI components to `UIConfig`

### Example of generic UI components related to the specific model:

We use `_ChatMessage` as a prefix since this component works with `_ChatMessage` model.
There's also a typeliases for `NoExtraData` case:

```swift
public typealias ChatMessageBubbleView = _ChatMessageBubbleView<NoExtraData>

open class _ChatMessageBubbleView<ExtraData: ExtraDataTypes>: View, UIConfigProvider { ... }
```

### Example of non-generic UI component related to the specific model:

We use `ChatMessage` as a prefix since this component works with `_ChatMessage` model. 
However, there's no `_` and typeliases since this type is not generic:

```swift
open class ChatMessageListCollectionView: UICollectionView { ... }
```

### Example of non-generic UI component NOT related to the specific model:

We use `Chat` as a prefix since to just avoid overlapping with `AvatarView` type declared in the end-users codebase. 
There's no `_` and alias since this type is not generic:

```swift
open class СhatAvatarView: View { ... }
```

### CODE REVIEW

File: Sources/StreamChatUI/ChatChannel/ChatMessage/ChatMessageGroupPart.swift [hunk 1]
```diff
@@ -4,15 +4,17 @@
...
-    public let didTapOnAttachment: ((_ChatMessageAttachment<ExtraData>) -> Void)?
-    public let didTapOnAttachmentAction: ((_ChatMessageAttachment<ExtraData>, AttachmentAction) -> Void)?
+    let message: _ChatMessage<ExtraData>
+    let quotedMessage: _ChatMessage<ExtraData>?
+    let isLastInGroup: Bool
+    let didTapOnAttachment: ((_ChatMessageAttachment<ExtraData>) -> Void)?
+    let didTapOnAttachmentAction: ((_ChatMessageAttachment<ExtraData>, AttachmentAction) -> Void)?
 
-    public var isPartOfThread: Bool {
+    var isPartOfThread: Bool {
```
Comment (VojtaStavik) [inline Sources/StreamChatUI/ChatChannel/ChatMessage/ChatMessageGroupPart.swift:17]: Why did you remove the `public` here?

File: Sources/StreamChatUI/ChatChannel/ChatMessage/ChatMessageCollectionViewCell.swift [hunk 2]
```diff
@@ -5,10 +5,13 @@
...
 
-open class СhatMessageCollectionViewCell<ExtraData: ExtraDataTypes>: UICollectionViewCell, UIConfigProvider {
+public typealias СhatMessageCollectionViewCell = _СhatMessageCollectionViewCell<NoExtraData>
+
+open class _СhatMessageCollectionViewCell<ExtraData: ExtraDataTypes>: CollectionViewCell, UIConfigProvider {
     class var reuseId: String { String(describing: self) }
+
     public var message: _ChatMessageGroupPart<ExtraData>? {
-        didSet { updateContent() }
+        didSet { updateContentIfNeeded() }
```
Comment (VojtaStavik) [inline Sources/StreamChatUI/ChatChannel/ChatMessage/ChatMessageCollectionViewCell.swift:14]: We need https://stream-io.atlassian.net/browse/CIS-535 😱

File: Sources/StreamChatUI/ChatChannel/MessageComposer/ChatMessageComposerCommandCollectionViewCell.swift [hunk 3]
```diff
@@ -78,15 +78,21 @@ open class _ChatMessageComposerCommandCellView<ExtraData: ExtraDataTypes>: View,
...
-    public private(set) lazy var commandView: _ChatMessageComposerCommandCellView<ExtraData> = {
-        let view = uiConfig.messageComposer.suggestionsCommandCellView.init().withoutAutoresizingMaskConstraints
-        contentView.embed(view)
-        return view
-    }()
+    override open func setUpLayout() {
+        super.setUpLayout()
+
+        contentView.embed(commandView)
+    }
```
Comment (VojtaStavik) [inline Sources/StreamChatUI/ChatChannel/MessageComposer/ChatMessageComposerCommandCollectionViewCell.swift]: 👍

File: Sources/StreamChatUI/Common Views/CurrentChatUserAvatarView.swift [hunk 4]
```diff
@@ -5,23 +5,24 @@
...
+public typealias CurrentChatUserAvatarView = _CurrentChatUserAvatarView<NoExtraData>
+
+open class _CurrentChatUserAvatarView<ExtraData: ExtraDataTypes>: Control, UIConfigProvider {
     // MARK: - Properties
     
     public var controller: _CurrentChatUserController<ExtraData>? {
         didSet {
             controller?.setDelegate(self)
-            updateContent()
+            updateContentIfNeeded()
```
Comment (VojtaStavik) [inline Sources/StreamChatUI/Common Views/CurrentChatUserAvatarView.swift:16]: 👍

File: Sources/StreamChatUI/Common Views/СhatAvatarView.swift [hunk 5]
```diff
@@ -4,7 +4,7 @@
 
 import UIKit
 
-open class AvatarView: View {
+open class СhatAvatarView: View {
```
Comment (VojtaStavik) [inline Sources/StreamChatUI/Common Views/СhatAvatarView.swift:7]: Was there "technically" a reason for changing this? 🤔

File: Sources/StreamChatUI/Common Views/ChatChannelCreateNewButton.swift [hunk 6]
```diff
@@ -17,14 +17,4 @@ open class CreateNewChannelButton: Button {
...
-    // MARK: - Init
-    
-    override public init(frame: CGRect) {
-        super.init(frame: frame)
-    }
-    
-    public required init?(coder: NSCoder) {
-        super.init(coder: coder)
-    }
 }
```
Comment (VojtaStavik) [inline Sources/StreamChatUI/Common Views/ChatChannelCreateNewButton.swift:20]: 👍

**Review by VojtaStavik — APPROVED** [review]
Awesome! Thank you for fixing so many tiny things when renaming 👍 🙏

File: Sources/StreamChatUI/Common Views/СhatAvatarView.swift (hunk 5 above)
Comment (evsaev) [reply Sources/StreamChatUI/Common Views/СhatAvatarView.swift:7]: Please refer to PR description, I explained why some types obtained `Chat` prefix

Comment (evsaev) [reply Sources/StreamChatUI/Common Views/СhatAvatarView.swift:7]: to me the reason for renaming `AvatarView` -> `ChatAvatarView` is the same as our decision to have `CurrentChatUser` instead of `CurrentUser`

File: Sources/StreamChatUI/ChatChannel/ChatMessage/ChatMessageGroupPart.swift (hunk 1 above)
Comment (evsaev) [reply Sources/StreamChatUI/ChatChannel/ChatMessage/ChatMessageGroupPart.swift:17]: Honestly, that's a surprise. I didn't manually change these access modifiers

Comment (evsaev) [reply Sources/StreamChatUI/ChatChannel/ChatMessage/ChatMessageGroupPart.swift]: fixed in 6e3e69bbc8ecb5eed62ba790dd05d226cf64444d
